.gitlab-ci.yml

```diff
@@ -12,3 +11,32 @@ build and push emulator image:
   script:
     - docker-compose -f ./docker-compose.yml build
     - docker-compose -f ./docker-compose.yml push
+
+build test:
+  stage: test
+  image: golang
+  variables:
+    TEST: "build-local"
+  script:
+    - go version
+    - go build .
+    - go test -v ./...
+
+go lint:
+  stage: test
+  image: golang
+  variables:
+    TEST: "golangci"
+  script:
+    - curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.32.0
+    - golangci-lint run
+
+version check:
+  stage: test
+  image: golang
+  variables:   
+    TEST: "version-check"
+  script:
+    - diff -u VERSION <(grep "var version = " trezord.go | cut -f 2 -d '"')
+
+#Gitlab does not support xgo building without hacking the base xgo image, for now builds are done maully until we find a better solution.
```

Comment:
Always add a newline at the end of a file please (yes, OCD thing :)).

Reply:
Ok, will fix it here :)